node-src/git/getParentCommits.ts

```diff
@@ -265,9 +257,40 @@ export async function getParentCommits(
     }
   );
 
+  const mergeInfoList = visitedCommitsWithoutBuilds.map((commit) => {
+    return { commit, baseRefName: branch };
+  });
+  const { app: { mergedPullRequests } } = await client.runQuery<MergeCommitsQueryResult>(
+    MergeCommitsQuery,
+    { mergeInfoList: mergeInfoList.slice(0, 100) }, // Limit amount sent in API call
+    { retries: 5 } // This query requires a request to an upstream provider which may fail
+  );
+
+  for (const pullRequest of mergedPullRequests) {
+    // Add the most recent build on a (merged) branch as an ancestor if we visit a commit
+    // during our ancestor selection that was the merge commit for that PR.
+    // @see https://www.chromatic.com/docs/branching-and-baselines#squash-and-rebase-merging
+    const lastHeadBuildCommit = pullRequest.lastHeadBuild?.commit;
+    if (lastHeadBuildCommit) {
+      if (await commitExists(lastHeadBuildCommit)) {
+        log.debug(
+          `Adding merged PR build commit ${lastHeadBuildCommit} to commits with builds`
+        );
+        commitsWithBuilds.push(lastHeadBuildCommit);
+      } else {
+        log.debug(
+          `Merged PR build commit ${lastHeadBuildCommit} not in index, blindly appending to parents`
+        );
+        extraParentCommits.push(lastHeadBuildCommit);
```

Comment:
The fact that we add the last head build to `commitsWithBuilds` and not just `extraParentCommits` is important because adding the build can "shadow" an existing "commit with build" which will no longer be used as an ancestor. Specifically, maybe we should add a test like:

```
      //     B - [D]       [branch]
       //    /
       // [A]- C -<E>-(G) [main]

```

And check that `A` is not used as ancestor as it is shadowed by `D`?